doc/adrs/183_order_representations_creation_modal.adoc

```diff
@@ -0,0 +1,62 @@
+= ADR-183 - Add support for ordering the representations in the creation modal
+
+== Context
+
+When creating a new representation from the UI, the user opens the `NewRepresentationModal` which proposes all the available representations which are compatible with the selected semantic target.
+In this modal, the available representations are obtained from the `viewer { editingContext { representationDescriptions() } }` query are presented [in alphabetical order of their label](https://github.com/eclipse-sirius/sirius-web/blob/master/packages/sirius-web/frontend/sirius-web-application/src/modals/new-representation/NewRepresentationModal.tsx#L256).
+
+While a good (predictable) default in the absence of more information, this alphabetical order is often not the most useful choice in a concrete modeling workbench, in particular when there is an "obvious" default choice of representation for the given target element.
+
+In addition, this alphabetical order may not be stable over time, if the representation's label is modified or if/when we support i18n for these labels.
+
+It should be possible to provide a custom order for these representations.
+
+== Decision
+
+We will provide a new extension point on the frontend with the following API:
```

Comment:
I think it would be better to do this on the backend. With a simple ordering service like we do with `IOmniboxCommandOrderer` (or like the explorer is ordering the documents for example). We can have more information in the backend to perform this kind of logic and it would make sure that this order is the same everywhere this information is requested.

I prefer the frontend not to make any assumptions.

Reply:
Done.